modules/tracking/include/opencv2/tracking/kalman_filters.hpp

```diff
@@ -222,6 +224,47 @@ CV_EXPORTS Ptr<UnscentedKalmanFilter> createUnscentedKalmanFilter( const Unscent
 */
 CV_EXPORTS Ptr<UnscentedKalmanFilter> createAugmentedUnscentedKalmanFilter( const AugmentedUnscentedKalmanFilterParams &params );
 
+
+namespace internal {
+
+/* Cholesky decomposition
+ The function performs Cholesky decomposition <https://en.wikipedia.org/wiki/Cholesky_decomposition>.
+ A - the Hermitian, positive-definite matrix,
+ astep - size of row in A,
+ asize - number of cols and rows in A,
+ L - the lower triangular matrix, A = L*Lt.
+*/
+template<typename _Tp> bool
+inline choleskyDecomposition( const _Tp* A, size_t astep, int asize, _Tp* L, size_t lstep)
+{
+    bool success = false;
+
+    astep /= sizeof(_Tp);
+    lstep /= sizeof(_Tp);
+
+    for(int i = 0; i < asize; i++)
+        for(int j = 0; j <= i; j++)
+            L[i*lstep + j] = A[i*astep + j];
+
+
+    if(typeid(_Tp) == typeid(float))
```

Comment:
you have this information already at compile time. use template specialization instead of RTTI.